cli/kata-check_amd64.go

```diff
@@ -7,52 +7,124 @@ package main
 
 import (
 	"github.com/sirupsen/logrus"
+	"io/ioutil"
+	"strings"
 )
 
 const (
 	cpuFlagsTag        = genericCPUFlagsTag
 	archCPUVendorField = genericCPUVendorField
 	archCPUModelField  = genericCPUModelField
+	archGenuineIntel   = "GenuineIntel"
+	archAuthenticAMD   = "AuthenticAMD"
+	msgKernelVM        = "Kernel-based Virtual Machine"
+	msgKernelVirtio    = "Host kernel accelerator for virtio"
+	msgKernelVirtioNet = "Host kernel accelerator for virtio network"
 )
 
+// CPU types
+const (
+	cPUtypeIntel   = 0
+	cPUtypeAMD     = 1
+	cPUtypeUnknown = -1
+)
+
+// cPUtype save the CPU type
+var cPUtype int
+
 // archRequiredCPUFlags maps a CPU flag value to search for and a
 // human-readable description of that value.
-var archRequiredCPUFlags = map[string]string{
-	"vmx":    "Virtualization support",
-	"lm":     "64Bit CPU",
-	"sse4_1": "SSE4.1",
-}
+var archRequiredCPUFlags map[string]string
 
 // archRequiredCPUAttribs maps a CPU (non-CPU flag) attribute value to search for
 // and a human-readable description of that value.
-var archRequiredCPUAttribs = map[string]string{
-	"GenuineIntel": "Intel Architecture CPU",
-}
+var archRequiredCPUAttribs map[string]string
 
 // archRequiredKernelModules maps a required module name to a human-readable
 // description of the modules functionality and an optional list of
 // required module parameters.
-var archRequiredKernelModules = map[string]kernelModule{
-	"kvm": {
-		desc: "Kernel-based Virtual Machine",
-	},
-	"kvm_intel": {
-		desc: "Intel KVM",
-		parameters: map[string]string{
-			"nested": "Y",
-			// "VMX Unrestricted mode support". This is used
-			// as a heuristic to determine if the system is
-			// "new enough" to run a Kata Container
-			// (atleast a Westmere).
-			"unrestricted_guest": "Y",
-		},
-	},
-	"vhost": {
-		desc: "Host kernel accelerator for virtio",
-	},
-	"vhost_net": {
-		desc: "Host kernel accelerator for virtio network",
-	},
+var archRequiredKernelModules map[string]kernelModule
+
+func setCPUtype() {
+	cPUtype = getCPUtype()
+
+	if cPUtype == cPUtypeUnknown {
+		exit(1)
```

Comment:
I think I'd be tempted to move the error log message from `getCPUtype()` to here as a quick glance at this code gives the impression this is a "bare exit".